### PR TITLE
Fix max issues and improve codecov

### PR DESCRIPTION
**Pull request by maximustan2**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -642,22 +642,22 @@ revert the change?<br>
...
 | Enroll Student            | `enroll m/MATRICULATION_NUMBER c/COURSE_CODE tut/TUTORIAL_ID`<br>e.g.`enroll m/A1234567Y c/CS1101S  tut/T10`                                                                                                           |
 | Unenroll Student          | `unenroll m/MATRICULATION_NUMBER c/COURSE_CODE tut/TUTORIAL_ID`<br>e.g.`unenroll m/A1234567Y c/CS1101S  tut/T10`                                                                                                       |
-| Mark Present           | `attend-present m/MATRICULATION_NUMBER c/COURSE_CODE tut/TUTORIAL_ID`<br>e.g.`attend attend m/A1234567Y c/CS1101S  tut/T10`                                                                                             |
+| Mark Present           | `attend-present m/MATRICULATION_NUMBER c/COURSE_CODE tut/TUTORIAL_ID`<br>e.g.`attend attend m/A1234567Y c/CS1101S  tut/T10`                                                                                            |
```
[the remark this answers is not in the document]
Comment (Zeepheru) [reply docs/UserGuide.md:660]: Yup, that looks right

File: src/test/java/tahub/contacts/storage/JsonStudentCourseAssociationListStorageTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        assertEquals(Optional.of(expectedList), actualList);
+    }
+
+    @Test
+    public void readScaList_invalidFile_throwsDataLoadingException() throws Exception {
+        storage = new JsonStudentCourseAssociationListStorage(INVALID_SCA_LIST_FILE);
+        assertThrows(
+                DataLoadingException.class, () -> storage.readScaList(INVALID_SCA_LIST_FILE, addressBook, courseList));
+    }
+}
```
Comment (Zeepheru) [inline src/test/java/tahub/contacts/storage/JsonStudentCourseAssociationListStorageTest.java:68]: Oooo that's pretty neat

**Review by Zeepheru — APPROVED** [review]
_LGTM_

File: docs/UserGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (FlyingDonkeys) [reply docs/UserGuide.md:660]: @miloaisdino the attend was repeated twice in the example, might have missed that out

File: src/main/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationList.java [hunk 3]
```diff
@@ -228,6 +228,14 @@ public void remove(StudentCourseAssociation toRemove) {
...
     }
 
+    /**
+     * Removes zero or more SCAs containing the specified course.
+     */
+    public void remove(Course course) {
+        requireNonNull(course);
+        internalList.removeIf(sca -> sca.getCourse().equals(course));
+    }
+
```
[the remark this answers is not in the document]
Comment (FlyingDonkeys) [reply src/main/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationList.java:238]: looks good

**Review by FlyingDonkeys — COMMENTED** [review]
generally looks ok

**Pull request merged**